src/InterNations/Sniffs/Syntax/MethodTypeHintsSniff.php

```diff
@@ -119,32 +102,33 @@ public function process(CodeSnifferFile $file, $stackPtr)
             return;
         }
 
-        // Ignore whitelisted methods by class
-        if (isset($this->ignoreTypeHintWhitelist[$className]) &&
-            in_array($methodName, $this->ignoreTypeHintWhitelist[$className])
-        ) {
-            return;
-        }
 
-        // Ignore whitelisted methods by parent class
-        if ($parentClassName &&
-            isset($this->ignoreTypeHintWhitelist[$parentClassName]) &&
-            in_array($methodName, $this->ignoreTypeHintWhitelist[$parentClassName])
-        ) {
-            return;
+        $namespace = '';
+        $namespaceTokenPtr = $file->findPrevious(T_NAMESPACE, $stackPtr - 1);
+        if ($namespaceTokenPtr) {
+            $namespacePtr = $file->findNext(T_STRING, $namespaceTokenPtr);
+            $endPtr = $file->findNext(T_SEMICOLON, $namespacePtr + 1);
+            $namespace =$file->getTokensAsString($namespacePtr, $endPtr - $namespacePtr) . '\\';
         }
 
-        // Ignore whitelisted methods by interface
-        if ($interfaces) {
-            foreach ($interfaces as $interface) {
-
-                if (isset($this->ignoreTypeHintWhitelist[$interface]) &&
-                    in_array($methodName, $this->ignoreTypeHintWhitelist[$interface])) {
+        try {
+            $reflectedClass = Util::reflectClass($namespace . $className);
+            do {
+                if ($this->isWhitelisted($reflectedClass, $methodName)) {
                     return;
                 }
-            }
+
+                foreach ($reflectedClass->getInterfaces() as $reflectedInterface) {
+                    if ($this->isWhitelisted($reflectedInterface, $methodName)) {
+                        return;
+                    }
+                }
+            } while ($reflectedClass = $reflectedClass->getParentClass());
```

Comment:
Super happy with this, very impressive thank you @lstrojny .  Now that, we have reflected class and can get parent class and interfaces, can we also achieve something like, if array, iterable etc.. params or return values are already defined in parent class and interface, we are allowed to skip in current child class, and phpcs should not complain?

Reply:
@CMalvika that should indeed work as well. I am not yet sure how fast it will be but that’s something we can figure out.